### ISSUE TITLE
Transferring missing legacy data to new slackmap

### ISSUE THREAD

**Issue by Can-Sahin** [International-Slackline-Association/slackmap#8]
We need to transfer the two things to new slackmap, which were forgotten during the migration
- Line types like Highline, Waterline
- Media files (photos)

We have the old data exported in json files and new data is also available, but they don't have matching IDs. So, there is a need for a script doing
1) Scanning all the lines and spots in the old JSON file and comparing the attributes like coordinates, description, name whatever with the new data and match the ID like OLD_1124123=NEW_12311231. So that we can keep a reference on which old data is referring to which new data
2) For each of the old lines and spots if there are images we need to scale them under 2MB limit
3) Write them into the new slackmap. This point is something I can do myself since it requires experience with the current system could be challenging for newcomers at first

**Comment by cseidholz** [comment]
Hi @Can-Sahin, I'd like to take a shot at this. Can you point me in the direction of the new and old datasets?

**Comment by Can-Sahin** [comment]
@tombucktwo

**Comment by cseidholz** [comment]
Thanks @Can-Sahin for the detailed instructions. I'm getting close to finishing and I noticed that the `description` field in the new dataset appears to have a 250 character limit. About ~50 entries in the old dataset had `description`s longer than that, causing the descriptions to become truncated in the new data. See [here](https://slackmap.com/spot/myUII_P) for an example of what I mean.

Do you want me to migrate the original (non-truncated) descriptions from the old dataset into the new one? I figured I'd ask first just in case the character limit was intentional.

**Comment by cseidholz** [comment]
A few more questions @Can-Sahin:

1) The old data contains both a `large` and a `small` version of each image (I'm assuming the small is a thumbnail). You mentioned wanting a `photosPath` field as an array of string paths to the images from the old data. Do you want to only include the large image in this array? Or would you want to include both images? We could make `photosPath` an array of objects, similar to how the old data is structured, like:
```json
"photosPath": [
      {
        "small": "data/photos/small/p033f83e758550d24d0af1c6142fc2dd7eb7cc5a19.jpg",
        "large": "data/photos/large/p033f83e758550d24d0af1c6142fc2dd7eb7cc5a19.jpg"
      } 
]
```

2) I was unable to match 37 out of the 8232 entries in the old data. From a brief look at the data, some of these seem like they might have intentionally been excluded from the new dataset (for instance - some of them are missing coordinate data, and a few others look like they were tests on the old slackmap). I will send these as a separate JSON if you want to see for yourself, but mostly I was just wondering - is this something you expected?

Besides these questions, I'm pretty much finished. Once I hear from you I can send the final JSON + pictures to you via email

**Comment by Can-Sahin** [comment]
Heyo
- Description is truncated but we have full version so don't worry about it. I'll add full version to db
- For the photos, small ones are pretty small so useless for us because currently no thumbnail needed. However large ones can be too large maybe (2mb +). So, only larges ones needed. Well if they are above 2mb then downscaled first. 
- 37 is pretty normal. I deleted some nonsense lines etc. And new file have several new spots we added later on. So all good

**Comment by Can-Sahin** [comment]
Oh another follow-up task is
- When the people visit old slackmap with ID in the path like, 'slackmap.com/x/ID' we should forward this request to 'slackmap.com/line/ID', or '/spot/ID'. 
- So we need to have a simple dictionary file hardcoded into the web app. So that we know where does this request are forwarded to which new path
```json
{
 "[OLD_ID]": {
   "type": 'spot' | 'line',
   "newId": 'XXX'
 },
....
}
```

**Comment by cseidholz** [comment]
Sweet thanks @Can-Sahin . I've finished up and have created the new line/spot dictionary. I'm gonna come back later tonight to write a little script to validate the data to be sure everything is good, then I'll plan on sending the data and images to you via email

**Comment by cseidholz** [comment]
Heya @Can-Sahin I sent the migrated data, photos, and a few scripts to you via email.

The final migrated data is `new.migrated.json`, the old data entries that could not be matched to new ones are in `old.unmatched.json` and the line/spot dictionary is `line-spot-dictionary.json`. 

I included a script to validate the migration and the photos - `validate-migration.js`. 

Note that all of the photos in `new.migrated.json` are currently pathed to the local `./data/photos` folder. I also included a script to make it easy to change those paths in bulk, just in case you're able to transfer the photos in bulk to the server.

Let me know if you have any questions on anything or if you find any mistakes. On a personal note I'm getting pretty stoked to put some work into this project so if you want to chat sometime about how I can best fit in, just shoot me an email. Otherwise I will just keep my eyes peeled for GH Issues.
